src/wasm/chaining_util.cpp

```diff
@@ -65,6 +65,10 @@ int makeChainedCall(const std::string& functionName,
         msg.set_issgx(true);
     }
 
+    if (originalCall->recordexecgraph()) {
+        msg.set_recordexecgraph(true);
+    }
+
```

Comment:
Need tests for both setting this to `true` in chained messages (should be adding a line to an existing test), and the call to `logChainedFunction` only when this is on.

As a general rule of thumb, if you're fixing a bug, you should be writing a test that would have caught it in the first place. Although this feature is still somewhat experimental, it's still very worth testing.